valet-key/ValetKey.Web/Controllers/HomeController.cs

```diff
@@ -52,36 +56,33 @@ public ActionResult Index()
 
         private async Task<StorageEntitySas> GetSharedAccessReferenceForDownload(string blobName)
         {
-            var blobClient = this.account.CreateCloudBlobClient();
-            var container = blobClient.GetContainerReference(this.blobContainer);
-
-            var blob = container.GetBlockBlobReference(blobName);
-            
-            if (!await blob.ExistsAsync())
-            {
-                throw new Exception("Blob does not exist");
-            }
+            var container = blobServiceClient.GetBlobContainerClient(this.blobContainer);
 
-            var policy = new SharedAccessBlobPolicy
-            {
-                Permissions = SharedAccessBlobPermissions.Read,
+            var blob = container.GetBlobClient(blobName);
 
-                // Create a signature for 5 min earlier to leave room for clock skew
-                SharedAccessStartTime = DateTime.UtcNow.AddMinutes(-5),
+            StorageSharedKeyCredential storageSharedKeyCredential = new StorageSharedKeyCredential(blobServiceClient.AccountName, "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==");
+       
+            var policy = new BlobSasBuilder
 
-                // Create the signature for as long as necessary -  we can 
-                SharedAccessExpiryTime = DateTime.UtcNow.AddMinutes(5)
+            {
+                Protocol = SasProtocol.None,
```

Comment:
Only include params that are required.  I think Protocol and IPRange are not required.